src/aggregator/aggregation/options.go

```diff
@@ -54,7 +60,21 @@ type GaugeMetrics struct {
 func NewMetrics(scope tally.Scope) Metrics {
 	scope = scope.SubScope("aggregation")
 	return Metrics{
-		Gauge: newGaugeMetrics(scope.SubScope("gauges")),
+		Counter: newCounterMetrics(scope.SubScope("counters")),
+		Gauge:   newGaugeMetrics(scope.SubScope("gauges")),
+	}
+}
+
+func newCounterMetrics(scope tally.Scope) CounterMetrics {
+	return CounterMetrics{
+		valuesOutOfOrder: scope.Counter("values-out-of-order"),
+	}
+}
+
+// IncValuesOutOfOrder increments value or if not initialized is a no-op.
+func (m CounterMetrics) IncValuesOutOfOrder() {
+	if m.valuesOutOfOrder != nil {
```

Comment:
Do we expect to initialize a `CounterMetrics` without the constructor or is this just to be paranoid?

Reply:
Bit of both, I believe some integration tests do not initialize it and also this is for paranoia.